### PR TITLE
Update Connection Resiliency docs

### LINKED ISSUE #405: Fix and expand the connection resiliency docs
For https://docs.microsoft.com/en-us/ef/core/miscellaneous/connection-resiliency

Add docs for `ExecuteInTransaction` overloads added in https://github.com/aspnet/EntityFramework/pull/8751 showing how to use the `verifySucceeded` parameter.
* The supplied operation lamda needs to be idempotent.
* Need to call `SaveChanges(acceptAllChangesOnSuccess: false)` as the transaction is committed outside of `SaveChanges` and might fail. Call `context.ChangeTracker.AcceptAllChanges()` after calling `ExecuteInTransaction`

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #405

### CODE REVIEW

File: entity-framework/core/miscellaneous/connection-resiliency.md [hunk 1]
```diff
@@ -10,10 +10,7 @@ uid: core/miscellaneous/connection-resiliency
 
 # Connection Resiliency
 
-> [!NOTE]  
-> This feature is new in EF Core 1.1.
-
-Connection resiliency automatically retries failed database commands. It is specific to relational databases.
+Connection resiliency automatically retries failed database commands. It is supported by all relational databases.
```
Comment (ajcvickers) [inline entity-framework/core/miscellaneous/connection-resiliency.md]: Does "supported" here imply that it works out-of-the-box for any relational database?

File: entity-framework/core/miscellaneous/connection-resiliency.md [hunk 2]
```diff
@@ -39,12 +36,57 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
+By default, the execution strategy will retry the operation as if the transaction was rolled back, but if it's not the case this will result in an exception if the new database state is incompatible or could lead to **data corruption** if the operation does not rely on a particular state, for example when inserting a new row with auto-generated key values.
+
+There are several ways to deal with this.
+
+### Option 1 - Do (almost) nothing
+
+The likelihood of a connection failure during transaction commit is low so it may be acceptable for your application to just fail if this condition actually occurs.
+
+However, you need to avoid using store-generated keys in order to ensure that an exception is thrown instead of adding a duplicate row. Consider using a client-generated GUID value or a client-side value generator such as `ForSqlServerUseSequenceHiLo`.
+
```
Comment (ajcvickers) [inline entity-framework/core/miscellaneous/connection-resiliency.md]: We shouldn't recommend people use ForSqlServerUseSequenceHiLo. It has other implications and is confusing for people who don't understand what is happening.

File: entity-framework/core/miscellaneous/connection-resiliency.md (hunk 1 above)
Comment (AndriySvyryd) [reply entity-framework/core/miscellaneous/connection-resiliency.md]: An execution strategy can be specified for any relational provider. But we only provide one for SQL Server

Comment (ajcvickers) [reply entity-framework/core/miscellaneous/connection-resiliency.md]: Yes, I understand that, I just think "supported" is ambiguous here. @divega can make the final call.

Comment (divega) [reply entity-framework/core/miscellaneous/connection-resiliency.md]: I would say something like connection resiliency _can_ be supported by any relational provider by implementing an execution strategy, and that the SQL Server provider does it.

Comment (divega) [reply entity-framework/core/miscellaneous/connection-resiliency.md]: I ended up coming up with something a bit more elaborate and more appropriate for the context (given the next section is on the SQL Server execution strategy). I will send a separate PR for review.

**Pull request merged**